compiler-core/src/type_/expression.rs

```diff
@@ -2918,17 +2916,19 @@ impl<'a, 'b> ExprTyper<'a, 'b> {
                         // If the function was valid but had the wrong number of arguments passed.
                         // Then we keep the error but still want to continue analysing the arguments that were passed.
                         MatchFunTypeError::IncorrectArity {
-                            args, return_type, ..
+                            args,
+                            return_type,
+                            expected,
+                            given,
+                            ..
                         } => {
+                            missing_args = expected.saturating_sub(given);
```

Comment:
Just wanted to get thoughts on this. Was about to start on https://github.com/gleam-lang/gleam/issues/3405 but I think this change here might just make this issue obsolete since if i'm reading this right you are adding the missing args in which would clear up any confusion?

Reply:
I only do that for use calls because I know the callback must be the last one, so I'm filling up the empty spaces with placeholder variables. For all other calls this wouldn't work so I'm guessing this doesn't solve the problem in that issue.
But now that I think about it we could do the same for calls with the wrong arity and labels to fill up the gaps and give better error messages, though it would be a little bit trickier to figure out where to put the placeholder args in case labels are used